docs/source/index.rst

```diff
@@ -0,0 +1,302 @@
+        ├── base.html
+        └── index.html
+
+base.html
+::
+
+    <!DOCTYPE html>
+    <html lang="en">
+        <head>
+            <meta charset="UTF-8">
+            <title>Flask-PluginEngine example</title>
+        </head>
+        <body>
+            {% block content %}
+            {% endblock %}
+        </body>
+    </html>
+
+index.html
+::
+
+    {% extends 'base.html' %}
+
+    {% block content %}
+        {% for plugin in plugins %}
+            <div>{{ plugin.title }} - {{ plugin.description }}</div>
+        {% endfor %}
+    {% endblock %}
+
+And let's pass the plugins, when rendering the template in our application::
+
+    @app.route("/")
+    def hello():
+        return render_template('index.html', plugins=active_plugins)
+
+Now what we should also do is to register the blueprints of all our plugins, for instance::
+
+    for plugin in active_plugins:
+        with plugin.plugin_context():
+            app.register_blueprint(plugin.get_blueprint())
+
+So our application code looks like this::
+
+    from flask import render_template
+    from flask_pluginengine import PluginFlask, PluginEngine
+
+    app = PluginFlask(__name__)
+
+    app.config['PLUGINENGINE_NAMESPACE'] = 'example'
+    app.config['PLUGINENGINE_PLUGINS'] = ['example_plugin']
+
+    plugin_engine = PluginEngine(app=app)
+    plugin_engine.load_plugins(app)
+    active_plugins = plugin_engine.get_active_plugins(app=app).values()
+
+    for plugin in active_plugins:
+        with plugin.plugin_context():
+            app.register_blueprint(plugin.get_blueprint())
+
+
+    @app.route("/")
+    def hello():
+        return render_template('index.html', plugins=active_plugins)
+
+
+    if __name__ == "__main__":
+        app.run()
+
+Now when we go to the index page of our application we can access the plugin's template.
+We can also directly access it if we go to `/plugin`.
+
+You can find the source code of the application in the `example folder <https://github.com/indico/flask-pluginengine/example/>`_.
+
+Configuring Flask-PluginEngine
+------------------------------
+
+The following configuration values exist for Flask-PluginEngine:
+
+====================================== ===========================================
+``PLUGINENGINE_NAMESPACE``             Specifies a namespace of the plugins
+``PLUGINENGINE_PLUGINS``               List of plugins the application will be
+                                       using
+====================================== ===========================================
+
+
+================
+PluginEngine API
+================
+
+
+.. automodule:: flask_pluginengine
+    :members: uses, depends, render_plugin_template, url_for_plugin
+
+PluginEngine
+------------
+
+.. autoclass:: PluginEngine
+    :members:
+
+Plugin
+------
+
+.. autoclass:: Plugin
+    :members: init
+
+    .. automethod:: plugin_context()
+    .. classmethod:: instance()
+
+        The Plugin instance used by the current app
+
+    .. classmethod:: title()
```

Comment:
Does it show up as `title()` or `title` in the docs? Because it's a class-level property and using `()` wouldn't work for obvious reasons ;)